interval.hh

```diff
@@ -426,7 +426,7 @@ std::ostream& operator<<(std::ostream& out, const wrapping_interval<U>& r) {
 }
 
 // An interval which can have inclusive, exclusive or open-ended bounds on each end.
-// The end bound can never be smaller than the start bound.
+// The end bound smaller than the start bound indicates an empty interval.
```

Comment:
I think it's not good to rely on this. Empty intervals can be created with the default constructor, and it should be illegal to call an interval with swapped parameters.